installer/scripts/service_control

```diff
@@ -59,6 +136,36 @@ setup_variables()
     OMSADMIN_CONF=$CONF_DIR/omsadmin.conf
 
     OMSAGENT_WS=omsagent-$WORKSPACE_ID
+
+    WS_STATUS=0
+}
+
+validate_omsadmin_conf()
+{
+    # Checks for a series of identifier assignment lines which should always be present
+    # for an onboarded workspace to be complete and healthy.
+    local _fSpec=$1
+    local _boolOnly=$2
+    VALIDATION_SUPPLEMENT_MESSAGE="Onboarding incorrect or incomplete."
+    if [ -f $_fSpec ]; then
+        validate_file_has_line_pattern $_fSpec "WORKSPACE_ID=$REGEX_UUID" $_boolOnly
+        validate_file_has_line_pattern $_fSpec "AGENT_GUID=$REGEX_UUID" $_boolOnly
+        validate_file_has_line_pattern $_fSpec "LOG_FACILITY=.*" $_boolOnly
+        validate_file_has_line_pattern $_fSpec "CERTIFICATE_UPDATE_ENDPOINT=.*" $_boolOnly
+        validate_file_has_line_pattern $_fSpec "URL_TLD=.*" $_boolOnly
+        validate_file_has_line_pattern $_fSpec "DSC_ENDPOINT=.*" $_boolOnly
+        validate_file_has_line_pattern $_fSpec "OMS_ENDPOINT=.*" $_boolOnly
+        validate_file_has_line_pattern $_fSpec "AZURE_RESOURCE_ID=.*" $_boolOnly
+        validate_file_has_line_pattern $_fSpec "OMSCLOUD_ID=.*" $_boolOnly
+        validate_file_has_line_pattern $_fSpec "UUID=$REGEX_UUID" $_boolOnly
+        return 0
+    else
+        if [ -z "$_boolOnly" ]; then
+            echo "INTERNAL_ERROR:  OMS Admin configuration file $_fSpec NOT Found!" >&2
+            echo $VALIDATION_SUPPLEMENT_MESSAGE
+        fi
+        exit $INTERNAL_ERROR
```

Comment:
Ok.  Sounds good.

Reply:
I think the way to go with this is leave the exits always on validation, and instead when multiples are handled, execute a subshell at that point so that you keep the solid consistency of a real validation routine, which by my experience exits on failure, and yet allows multiple processing by use of the subprocess that can just do the exiting without leaving the entire set.  This makes the code cleaner too.

Reply:
That design is up to you. If you can implement it and test it successfully, then that sounds good.